Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp

```diff
@@ -0,0 +1,255 @@
+    int idx = 0;
+    while (mediaDescriptionIter.HasCurrent())
+    {
+
+        auto format = mediaDescriptionIter.Current();
+        int frameRateInt = (int)format.FrameRate();
+        int frameRateDec = ((int)(format.FrameRate() * 100) % 100);
+        idx++;
+        auto formatStr = std::to_wstring(idx) + L". " + std::wstring(format.Subtype().c_str()) + L": " +
+            std::to_wstring(format.Width()) + L"x" + std::to_wstring(format.Height()) + L" @ " + std::to_wstring(frameRateInt) + L"." + std::to_wstring(frameRateDec) + L" fps";
+        std::wcout << formatStr << std::endl;
+        mediaDescriptionIter.MoveNext();
+    }
+    int selection;
+    do
+    {
+        std::wcout << L"Enter selection:";
+        std::cin >> selection;
+    } while ((selection > idx) || (selection < 1));
+    return selection-1; // because selection idx is from 1 to N and indexes for from 0 to N-1
+}
+
+void TakePhotosAndProcess(MediaCapture mediaCapture, uint32_t photoIndex)
+{
+    auto folderRoot = StorageFolder::GetFolderFromPathAsync(L"C:\\").get();
+    auto folder = folderRoot.CreateFolderAsync(L"test\\", CreationCollisionOption::OpenIfExists).get();
+    auto file1 = folder.CreateFileAsync(to_hstring(photoIndex) + L"_1.png", CreationCollisionOption::ReplaceExisting).get();
+    auto file2 = folder.CreateFileAsync(to_hstring(photoIndex) + L"_2.png", CreationCollisionOption::ReplaceExisting).get();
+
+    //Capture and save two photos
+    mediaCapture.CapturePhotoToStorageFileAsync(ImageEncodingProperties::CreatePng(), file1).get();
+    mediaCapture.CapturePhotoToStorageFileAsync(ImageEncodingProperties::CreatePng(), file2).get();
+
+    auto sbList = single_threaded_vector<SoftwareBitmap>();
+    // open photos and decode them into NV12 format for processing
+    // photo1
+    {
+        auto strm = file1.OpenReadAsync().get();
+        auto decoder = BitmapDecoder::CreateAsync(strm).get();
+        auto sb1 = decoder.GetSoftwareBitmapAsync().get();
+        sbList.Append(SoftwareBitmap::Convert(sb1, BitmapPixelFormat::Nv12));
+    }
+    // photo2
+    {
+        auto strm = file2.OpenReadAsync().get();
+        auto decoder = BitmapDecoder::CreateAsync(strm).get();
+        auto sb1 = decoder.GetSoftwareBitmapAsync().get();
+        sbList.Append(SoftwareBitmap::Convert(sb1, BitmapPixelFormat::Nv12));
+    }
+
+    // Do some processing like low light fusion in this example
+    auto op = LowLightFusion::FuseAsync(sbList).get();
+
+    // Save the output
+    auto fname = to_hstring(photoIndex) + L"_LLF.png";
+    auto fileOp = folder.CreateFileAsync(fname, CreationCollisionOption::ReplaceExisting).get();
+    {
+        auto strm = fileOp.OpenAsync(FileAccessMode::ReadWrite).get();
+        auto encoder = BitmapEncoder::CreateAsync(BitmapEncoder::PngEncoderId(), strm).get();
+        encoder.SetSoftwareBitmap(SoftwareBitmap::Convert(op.Frame(), BitmapPixelFormat::Bgra8));
+        encoder.FlushAsync().get();
+    }
+}

```

Comment:
nit: Maybe add a couple of output altering the user of the progress of the steps, i.e. "capturing photo 1", "capturing photo 2", "fusing both photos", "low light fusion completed, all done!"

Reply:
adding file paths as well in that debug output would be nice

---
In reply to: [284525919](https://github.com/microsoft/Windows-Camera/pull/9#discussion_r284525919) [](ancestors = 284525919)